Add support for rpm based installs

scollazo opened the pull request:
This is a refactor/cleanup of #252 

The only change needed to install from rpm's is:

    archivematica_src_install_ss: "rpm"
    archivematica_src_install_am: "rpm"

scollazo committed:
Install p7zip-plugins

The package includes /usr/bin/7zip, no need to create a link

scollazo committed:
Add support for rpm (common)  [1/3]

 - Add rpm repositories to vars file
 - Split pip install into block
 - Added block to common for repository configuration
 - Removed visudo tasks, we are not using sudo since some time ago

In order to install from rpm's, archivematica_src_install_ss/am needs to be
configured as follows:

archivematica_src_install_ss: "rpm"
archivematica_src_install_am: "rpm"

Using "true" or "yes" will install from sources

scollazo committed:
Add support for rpm (ss) [2/3]

 - Add support for ss rpm packages
 - Split common tasks needed in source and rpm installs into ss-osconf
 - Removes gunicorn conf file provided by the rpm

scollazo committed:
Add support for rpm installs (dashboard) [3/3]

 - pipeline-instcode only installs archivematica code
 - removed logging configuration from pipeline-osconf
 - gunicorn and services are now in pipeline-environment, instead
 of pipeline-websrv-gunicorn

Ideally, we should be able to reconfigure an instance without the need of
reinstalling it.

ross-spencer reviewed (commented):
Thank Santi. This is really useful to read through, I've some follow-up reading to do now too.

mamedin commented on tasks/common.yml:
```diff
@@ -42,40 +16,75 @@
+      description: "{{ item.key }}"
+      baseurl: "{{ item.value.baseurl }}"
+      repo_gpgcheck: "{{ item.value.gpgcheck }}"
+    loop: "{{ lookup('dict',archivematica_src_rpm_repositories) }}"
```
The repo needs the gpgkey url, I got the error:

```
TASK [external-roles/artefactual-archivematica-qa-src : Install necessary packages required by this ansible role] *************************************************************************************************
[DEPRECATION WARNING]: Invoking "yum" only once while using a loop via squash_actions is deprecated. Instead of using a loop to supply multiple items and specifying `name: "{{ item }}"`, please use `name: 
['libsemanage-python', 'policycoreutils-python']` and remove the loop. This feature will be removed in version 2.11. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
failed: [mamedin-ansible-rpms] (item=[u'libsemanage-python', u'policycoreutils-python']) => {"ansible_loop_var": "item", "changed": false, "item": ["libsemanage-python", "policycoreutils-python"], "msg": "https://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica\nTrying other mirror.\n\n\n One of the configured repositories failed (archivematica),\n and yum doesn't have enough cached data to continue. At this point the only\n safe thing yum can do is fail. There are a few ways to work \"fix\" this:\n\n     1. Contact the upstream for the repository and get them to fix the problem.\n\n     2. Reconfigure the baseurl/etc. for the repository, to point to a working\n        upstream. This is most often useful if you are using a newer\n        distribution release than is supported by the repository (and the\n        packages for the previous distribution release still work).\n\n     3. Run the command with the repository temporarily disabled\n            yum --disablerepo=archivematica ...\n\n     4. Disable the repository permanently, so yum won't use it by default. Yum\n        will then just ignore the repository until you permanently enable it\n        again or use --enablerepo for temporary usage:\n\n            yum-config-manager --disable archivematica\n        or\n            subscription-manager repos --disable=archivematica\n\n     5. Configure the failing repository to be skipped, if it is unavailable.\n        Note that yum will try to contact the repo. when it runs most commands,\n        so will have to try and fail each time (and thus. yum will be be much\n        slower). If it is a very temporary problem though, this is often a nice\n        compromise:\n\n            yum-config-manager --save --setopt=archivematica.skip_if_unavailable=true\n\nfailure: repodata/repomd.xml from archivematica: [Errno 256] No more mirrors to try.\nhttps://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica\n", "rc": 1, "results": []}
```
When running `sudo yum repolist` on CentOS VM:

```
https://packages.archivematica.org/1.10.x/centos/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica
...
https://packages.archivematica.org/1.10.x/centos-extras/repodata/repomd.xml: [Errno -1] Gpg Keys not imported, cannot verify repomd.xml for repo archivematica-extras
``` 
Taking a look to the archivematica-docs CentOS install script ( https://raw.githubusercontent.com/artefactual/archivematica-docs/1.10/admin-manual/installation-setup/installation/scripts/am-centos-rpm.sh ) this task needs to add to both yum repo files:

```
gpgkey=https://packages.archivematica.org/1.10.x/key.asc
```

scollazo replied:
Added gpgkey and gpgcheck to the configuration dict.

scollazo committed:
rpm: configure gpg keys

Also, am packages are now updated always

scollazo committed:
Change include: into include_tasks:

The use of 'static' has been deprecated. Ansible recommends
'import_tasks' for static inclusion, or 'include_tasks'
for dynamic inclusion

mamedin commented:
The 7z symlink is missing on CentOS deploy with rpms.  When testing an ingest I got the error on `Prepare AIP> compress AIP` microservice:

``` 
/tmp/f706a058-3bb3-4500-bfaa-54c5f6814b25: line 2: /usr/bin/7z: No such file or directory
/tmp/cadcc06d-c948-457e-b960-22b8fd69d2b6: line 2: 7z: command not found
``` 

See https://www.archivematica.org/en/docs/archivematica-1.10/admin-manual/installation-setup/installation/install-centos/#install-pkg-centos :

``` 
sudo ln -sf /usr/bin/7za /usr/bin/7z
``` 

I'm going to test in CentoS deploy from sources.

mamedin commented:
Hi again Santi, as you suggested on slack it is related with the `p7zip-plugins` package. 

CentOS rpm deploy:
```
[artefactual@mamedin-ansible-rpms ~]$ sudo rpm -qa | grep 7z
p7zip-16.02-10.el7.x86_64
```
CentOS deploy from sources:
``` 
[artefactual@mamedin-ansible-rpms-centos-src ~]$ sudo rpm -qa | grep p7
p7zip-plugins-16.02-10.el7.x86_64
p7zip-16.02-10.el7.x86_64
```

mamedin reviewed (approved):
Good job!!
